Спринт 11

KateSilencio opened the pull request:
Работа с сетью
Версия 1

KateSilencio committed:
Спринт 11
Работа с сетью
Версия 1

askhad-apishev commented on app/src/main/res/drawable/ic_nothing_found_light.xml:
```diff
@@ -0,0 +1,20 @@
+      android:fillColor="#FCD452"/>
+  <path
+      android:pathData="M111,6L105,18.21V79.3C102.38,76.61 99.08,75 95.5,75C86.94,75 80,84.18 80,95.5C80,106.82 86.94,116 95.5,116C103.68,116 110.38,107.62 110.96,97H111V95.5V6Z"
+      android:fillColor="#1A1B22"
```
цвета должны браться из ресурсов

askhad-apishev commented on app/src/main/res/drawable/ic_nothing_found_dark.xml:
```diff
@@ -0,0 +1,22 @@
+      android:fillColor="#FCD452"/>
+  <path
+      android:pathData="M80,95.5a15.5,20.5 0,1 0,31 0a15.5,20.5 0,1 0,-31 0z"
+      android:fillColor="#ffffff"/>
```
цвета должны браться из ресурсов

askhad-apishev commented on app/src/main/java/com/example/playlistmaker/data/ITunesResponse.kt:
```diff
@@ -0,0 +1,20 @@
+    @SerializedName("results")
+    val results: List<ResultFields>
+)
+data class ResultFields(
```
очень странное имя для класса, которое не сообщать никакой информации о нем.
`Track`

askhad-apishev commented on app/src/main/java/com/example/playlistmaker/data/ITunesResponse.kt:
```diff
@@ -0,0 +1,20 @@
+    @SerializedName("resultCount")
+    val resultCount: Int,
+    @SerializedName("results")
+    val results: List<ResultFields>
```
можно для удобства использовать более осмысленное имя `tracks` 🙃

askhad-apishev commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,68 +1,51 @@
 
+    private val trackList = ArrayList<ResultFields>()
+    companion object {
         private const val EDIT_TEXT = "EditText"
```
более говорящим именем будет `SEARCH_QUERY_KEY` т к это ключ для чтения/записи поискового запроса пользователя

askhad-apishev commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,68 +1,51 @@
+    private val trackList = ArrayList<ResultFields>()
+    companion object {
         private const val EDIT_TEXT = "EditText"
+        private var edit = ""
```
можно создать константу `EMPTY` и использовать везде ее, где нужна пустая строчка

askhad-apishev commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,68 +1,51 @@
+    companion object {
         private const val EDIT_TEXT = "EditText"
+        private var edit = ""
+        //строка для сохранения последнего запроса
```
не нужно захламлять код этими комментариями. название переменной должно говорить само за себя. если возникает необходимость в поясняющем комментарии, то нужно подумать над более лучшим названием переменной

askhad-apishev commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -96,34 +82,100 @@ class SearchActivity : AppCompatActivity() {
-    fun getRandomTrack(): Track = trackList[(0..4).random()]
+    //Функция обновления списка в адаптере и установка видимости контейнеров
+    @SuppressLint("NotifyDataSetChanged")
+    private fun updateListTracks(data: List<ResultFields>, boolArray: Array<Boolean>){
```
не очень хорошая практика использовать параметр `boolArray` для регулирования видимости. лучше все разбить на отдельные методы. 

p.s. к тому же, непонятно как пользоваться этим параметром. что значит false, true, false? a что значит когда все false или все true? это "типо оптимизация", но нет 🙃 

у каждого состояния должно быть свое имя. в данном случае это имя отдельных функций. в будущем это могут быть названия sealed классов

askhad-apishev commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -96,34 +82,100 @@ class SearchActivity : AppCompatActivity() {
+                        }
+                    } else {
+                        //Неудачный запрос
+                        updateListTracks(emptyList(), arrayOf(false,false,true))     //??? стоит ли обновлять список пустым списком ???
```
как будет понятней: когда это arrayOf(false, false, true) или когда мы вызовем функцию `onFailureRequest()` ?

askhad-apishev commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -96,34 +82,100 @@ class SearchActivity : AppCompatActivity() {
+                            Log.d("SUCCESS", "Успех ${response.code()}")
+                        } else {
+                            //результат запроса пустой:
+                            updateListTracks(emptyList(), arrayOf(false,true,false))     //??? стоит ли обновлять список пустым списком ???
```
как будет понятней: когда это arrayOf(false, true, false) или когда мы вызовем функцию `onEmptyResponse()` ?

askhad-apishev commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -96,34 +82,100 @@ class SearchActivity : AppCompatActivity() {
+
+                        if (result.isNotEmpty()){
+                            //результат запроса не пустой:
+                            updateListTracks(result, arrayOf(true,false,false))
```
как будет понятней: когда это arrayOf(true, false, false) или когда мы вызовем функцию `onSuccess()` ?

KateSilencio committed:
Спринт 11 Работа с сетью
Версия 2
Исправлено:
1.добавлен атрибут background ко всем activity.xml
2.в track_view.xml добавлено:
- доп ConstraintLayout для artist_name, dot, track_time
-доп привязки к краям и атрибуты
3. Добавлены цвета из ресурсов в иконки
4. Переименованы названия переменных и классов
5. доб ф-ии управления видимостью контейнеров
